File: ios/ChenYuanSite/UI/Theme.swift
import SwiftUI

/// 站点黑白灰单色体系（DESIGN.md 令牌），深浅色跟随系统：
/// 亮色 ink #1c1c1e / surface #ffffff / muted #656568 / line #eeeeee；
/// 暗色背景 #181818 / 前景 #f4f4f4。无卡片、无重阴影，层级靠细线、留白与字重。
enum SiteTheme {
    nonisolated static let background = dynamic(light: 0xFFFFFF, dark: 0x181818)
    nonisolated static let ink = dynamic(light: 0x1C1C1E, dark: 0xF4F4F4)
    nonisolated static let muted = dynamic(light: 0x656568, dark: 0xA1A1A4)
    nonisolated static let quiet = dynamic(light: 0x767676, dark: 0x8E8E93)
    nonisolated static let line = dynamic(light: 0xEEEEEE, dark: 0x2A2A2C)
    /// 玻璃底色：亮色白 80% 透明，暗色 #181818 70% 透明（对应安卓 glass 令牌）。
    nonisolated static let glass = Color(uiColor: UIColor { trait in
        trait.userInterfaceStyle == .dark
            ? UIColor(red: 0x18 / 255, green: 0x18 / 255, blue: 0x18 / 255, alpha: 0.70)
            : UIColor(red: 1, green: 1, blue: 1, alpha: 0.80)
    })

    private nonisolated static func dynamic(light: UInt32, dark: UInt32) -> Color {
        Color(uiColor: UIColor { trait in
            UIColor(hex: trait.userInterfaceStyle == .dark ? dark : light)
        })
    }
}

private extension UIColor {
    nonisolated convenience init(hex: UInt32) {
        self.init(
            red: CGFloat((hex >> 16) & 0xFF) / 255,
            green: CGFloat((hex >> 8) & 0xFF) / 255,
            blue: CGFloat(hex & 0xFF) / 255,
            alpha: 1,
        )
    }
}

/// 站点排版阶梯的 App 侧映射（SiteText）：标题 600 字重，正文 15，辅助 12。
/// iOS 无独立 lineHeight 概念，用 lineSpacing 逼近安卓的行高比。
enum SiteText {
    static let identity = Font.system(size: 32, weight: .medium)
    static let pageTitle = Font.system(size: 22, weight: .semibold)
    static let title = Font.system(size: 16, weight: .medium)
    static let listTitle = Font.system(size: 15, weight: .medium)
    static let body = Font.system(size: 15)
    static let summary = Font.system(size: 13)
    static let meta = Font.system(size: 12)
    static let eyebrow = Font.system(size: 12, weight: .medium)
    static let label = Font.system(size: 14, weight: .medium)
    static let tab = Font.system(size: 14)
    static let tabSelected = Font.system(size: 14, weight: .semibold)

    // 行距修正：安卓 15/26 ≈ +6，13/21 ≈ +3，12/18 ≈ +2。
    static let bodyLineSpacing: CGFloat = 6
    static let summaryLineSpacing: CGFloat = 3
    static let metaLineSpacing: CGFloat = 2
}

/// 间距令牌（SiteSpace）。
enum SiteSpace {
    static let micro: CGFloat = 4
    static let compact: CGFloat = 8
    static let related: CGFloat = 12
    static let paragraph: CGFloat = 16
    static let item: CGFloat = 20
    static let page: CGFloat = 24
    static let section: CGFloat = 24
    static let touch: CGFloat = 48
}

// 全站通用文本修饰：一处定义行距，避免每个 Text 手写。
extension View {
    func siteBodyStyle(_ color: Color = SiteTheme.ink) -> some View {
        font(SiteText.body).lineSpacing(SiteText.bodyLineSpacing).foregroundStyle(color)
    }

    func siteSummaryStyle(_ color: Color = SiteTheme.muted) -> some View {
        font(SiteText.summary).lineSpacing(SiteText.summaryLineSpacing).foregroundStyle(color)
    }

    func siteMetaStyle(_ color: Color = SiteTheme.quiet) -> some View {
        font(SiteText.meta).lineSpacing(SiteText.metaLineSpacing).foregroundStyle(color)
    }
}

File: android/app/src/main/java/cn/lovemyrmb/personalsite/ui/theme/Theme.kt
package cn.lovemyrmb.personalsite.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Typography
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/**
 * 站点黑白灰单色体系（DESIGN.md 令牌）：亮色 ink #1c1c1e / surface #ffffff /
 * muted #656568 / line #eeeeee；暗色背景 #181818 / 前景 #f4f4f4。
 * 无卡片、无重阴影，层级靠细线、留白与字重。
 */
@Immutable
data class SiteColors(
    val background: Color,
    val ink: Color,
    val muted: Color,
    val quiet: Color,
    val line: Color,
    val glass: Color,
)

private val LightSiteColors = SiteColors(
    background = Color(0xFFFFFFFF),
    ink = Color(0xFF1C1C1E),
    muted = Color(0xFF656568),
    quiet = Color(0xFF767676),
    line = Color(0xFFEEEEEE),
    glass = Color(0xCCFFFFFF),
)

private val DarkSiteColors = SiteColors(
    background = Color(0xFF181818),
    ink = Color(0xFFF4F4F4),
    muted = Color(0xFFA1A1A4),
    quiet = Color(0xFF8E8E93),
    line = Color(0xFF2A2A2C),
    glass = Color(0xB3181818),
)

val LocalSiteColors = staticCompositionLocalOf { LightSiteColors }

object SiteTheme {
    val colors: SiteColors
        @Composable get() = LocalSiteColors.current
}

// 站点排版阶梯的 App 侧映射：标题 600 字重，正文 15sp，辅助 12sp。
object SiteText {
    private val base = TextStyle(fontFamily = FontFamily.SansSerif, letterSpacing = 0.sp, fontWeight = FontWeight.Normal)
    val identity = base.copy(fontSize = 32.sp, lineHeight = 40.sp, fontWeight = FontWeight.Medium)
    val pageTitle = base.copy(fontSize = 22.sp, lineHeight = 30.sp, fontWeight = FontWeight.SemiBold)
    val title = base.copy(fontSize = 16.sp, lineHeight = 24.sp, fontWeight = FontWeight.Medium)
    val listTitle = base.copy(fontSize = 15.sp, lineHeight = 22.sp, fontWeight = FontWeight.Medium)
    val body = base.copy(fontSize = 15.sp, lineHeight = 26.sp)
    val summary = base.copy(fontSize = 13.sp, lineHeight = 21.sp)
    val meta = base.copy(fontSize = 12.sp, lineHeight = 18.sp)
    val eyebrow = meta.copy(fontWeight = FontWeight.Medium)
    val label = base.copy(fontSize = 14.sp, lineHeight = 20.sp, fontWeight = FontWeight.Medium)
    val tab = label.copy(fontWeight = FontWeight.Normal)
    val tabSelected = label.copy(fontWeight = FontWeight.SemiBold)
}

object SiteSpace {
    val micro = 4.dp
    val compact = 8.dp
    val related = 12.dp
    val paragraph = 16.dp
    val item = 20.dp
    val page = 24.dp
    val section = 24.dp
    val touch = 48.dp
}

private val SiteTypography = Typography(
    displayLarge = SiteText.identity, displayMedium = SiteText.identity, displaySmall = SiteText.identity,
    headlineLarge = SiteText.identity, headlineMedium = SiteText.pageTitle, headlineSmall = SiteText.pageTitle,
    titleLarge = SiteText.title, titleMedium = SiteText.listTitle, titleSmall = SiteText.label,
    bodyLarge = SiteText.body, bodyMedium = SiteText.summary, bodySmall = SiteText.meta,
    labelLarge = SiteText.label, labelMedium = SiteText.eyebrow, labelSmall = SiteText.meta,
)

private val LightScheme = lightColorScheme(
    background = LightSiteColors.background,
    onBackground = LightSiteColors.ink,
    surface = LightSiteColors.background,
    onSurface = LightSiteColors.ink,
    surfaceVariant = LightSiteColors.line,
    onSurfaceVariant = LightSiteColors.muted,
    outline = LightSiteColors.line,
    primary = LightSiteColors.ink,
    onPrimary = LightSiteColors.background,
)

private val DarkScheme = darkColorScheme(
    background = DarkSiteColors.background,
    onBackground = DarkSiteColors.ink,
    surface = DarkSiteColors.background,
    onSurface = DarkSiteColors.ink,
    surfaceVariant = DarkSiteColors.line,
    onSurfaceVariant = DarkSiteColors.muted,
    outline = DarkSiteColors.line,
    primary = DarkSiteColors.ink,
    onPrimary = DarkSiteColors.background,
)

@Composable
fun PersonalSiteTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit,
) {
    val colors = if (darkTheme) DarkSiteColors else LightSiteColors
    CompositionLocalProvider(LocalSiteColors provides colors) {
        MaterialTheme(
            typography = SiteTypography,
            colorScheme = if (darkTheme) DarkScheme else LightScheme,
            content = content,
        )
    }
}
